src/LatteContext/Collector/FormFieldCollector.php

```diff
@@ -103,26 +104,39 @@ public function collectData(Node $node, Scope $scope): ?array
             }
 
             $formFieldType = $formFieldParametersAcceptor->getReturnType();
-
             $fieldNameArg = $node->getArgs()[0] ?? null;
-            $fieldNameDefaultType = $formFieldParametersAcceptor->getParameters()[0]->getDefaultValue();
+
+            $formFieldParameters = $formFieldParametersAcceptor->getParameters();
+            $fieldNameDefaultType = isset($formFieldParameters[0]) ? $formFieldParameters[0]->getDefaultValue() : null;
             if ($fieldNameDefaultType instanceof ConstantStringType) {
                 $fieldNameDefault = trim($fieldNameDefaultType->getValue(), '"\'');
             } else {
                 $fieldNameDefault = null;
             }
         }
 
-        if (!$formFieldType instanceof ObjectType) {
-            return null;
+        $formFieldTypes = [];
```

Comment:
My method addAccount() has union return type, so I'm trying to solve it here. It works like expceted, I just don't like the code. Any idea how to write it more phpstan-like?

Reply:
Try TypeCombinator::intersect return type with union of allowed types.

Reply:
I'll play with it later, now I'm going to merge